app/src/main/java/com/example/harvestfresh/DetailActivity.java

```diff
@@ -2,15 +2,80 @@
 
 import android.os.Bundle;
 import android.os.PersistableBundle;
+import android.widget.ImageView;
+import android.widget.TextView;
 
 import androidx.annotation.Nullable;
 import androidx.appcompat.app.AppCompatActivity;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.bumptech.glide.Glide;
+import com.parse.FindCallback;
+import com.parse.ParseException;
+import com.parse.ParseFile;
+import com.parse.ParseQuery;
+import com.parse.ParseUser;
+
+import org.parceler.Parcels;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
 
 public class DetailActivity extends AppCompatActivity {
 
+    private static int PRODUCT_LIMIT = 20;
+
+    private RecyclerView rvProduct;
+    private ProductListingAdapter fragmentAdapter;
+    private List<ProductListing> allProducts;
+
+    StoreFront store;
+    TextView tvStoreName;
+    ImageView ivStoreImage;
+
     @Override
     protected void onCreate(@Nullable Bundle savedInstanceState) {
```

Comment:
This method is doing a lot of different things...something we can do for legibility is breaking it up into a handful of smaller methods. For example
onCreate(){
androidBoilerplate(); // pick a better name for this

store = blah;
setStoreNameText(store);
setStoreImage(store);

setupProductRecyclerView();
queryProduct();
}

You can see how I took the 4-5 different things this method tries to do, and put them into their own methods. This way, I can quickly enter the class, look at the entry method, and find where I want to make my code change in the future. If I want to change the image code, I dont need to read through all the text code and boilerplate to find it, and I have a guarantee that there isn't spaghetti code somewhere else that relates to images if theres one image method. Similarly, it makes the code reusable, so it would be easy to call setStoreNameText() from somewhere else if we wanted to load a different store or something.